### PR TITLE
Feat/akita

### PR DESCRIPTION
**Pull request by lfa-vasconcellos**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2103,6 +2103,16 @@
         "to-fast-properties": "^2.0.0"
       }
     },
+<<<<<<< HEAD
```
Comment (Artfloriani) [inline package-lock.json:2106]: Please remove merge conflicts

File: src/app/app.module.ts [hunk 2]
```diff
@@ -3,21 +3,29 @@ import { BrowserModule } from '@angular/platform-browser';
...
+import { StorageService } from './core/services/storage/storage.service';
 
 @NgModule({
   declarations: [AppComponent],
   entryComponents: [],
-  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule],
+  imports: [
+    BrowserModule,
+    IonicModule.forRoot(),
+    IonicStorageModule.forRoot(),
```
Comment (Artfloriani) [inline src/app/app.module.ts:21]: Where are you setting IonicStorage as the default storage for Akita? 

If you want to use Ionic as the default, you need to explicitly set it: https://datorama.github.io/akita/docs/enhancers/persist-state

The problem is that IonicStorage does not implement the StorageEngine API, so we would need a wrapper for that :(

File: src/app/modules/example-feature/example.interface.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export interface ExampleInterface {
```
Comment (Artfloriani) [inline src/app/modules/example-feature/example.interface.ts:1]: I think this should be moved inside the `state` folder

File: src/app/modules/example-feature/state/examples.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+  loadExamples(): Observable<ExampleInterface[]> {
+    // On a real service should call an API service method
+    const example: ExampleInterface = { id: 1, label: 'example' };
+
+    return of([example]).pipe(
+      tap(response => {
+        // Set it in store
+        // this.examplesStore.set(response);
```
Comment (Artfloriani) [inline src/app/modules/example-feature/state/examples.service.ts:24]: You can set it here as an example and use this value on the the example page

File: src/app/modules/example-feature/example.interface.ts (hunk 3 above)
Comment (lfa-vasconcellos) [reply src/app/modules/example-feature/example.interface.ts:1]: Done

File: src/app/modules/example-feature/state/examples.service.ts (hunk 4 above)
Comment (lfa-vasconcellos) [reply src/app/modules/example-feature/state/examples.service.ts:24]: If we set this as an example, will it not pollute the local storage unnecessarily?

File: package-lock.json (hunk 1 above)
Comment (lfa-vasconcellos) [reply package-lock.json:2106]: Done

File: src/app/core/services/storage/storage.service.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { enableAkitaProdMode, persistState } from '@datorama/akita';
+import { environment } from '../../../../environments/environment.prod';
```
Comment (Ryssa) [inline src/app/core/services/storage/storage.service.ts:3]: Environment imports should be from `'../../../../environments/environment`.

File: src/app/modules/example-feature/components/presentation/presentation.component.ts [hunk 6]
```diff
@@ -1,10 +1,12 @@
 import { Component, OnInit } from '@angular/core';
+import { ExamplesService } from '../../state/examples.service';
 
 @Component({
   selector: 'app-presentation',
   templateUrl: './presentation.component.html',
   styleUrls: ['./presentation.component.scss'],
 })
 export class PresentationComponent implements OnInit {
+  constructor(private exampleService: ExamplesService) { this.exampleService.loadExamples().subscribe(s => console.log(s)); }
```
Comment (Ryssa) [inline src/app/modules/example-feature/components/presentation/presentation.component.ts:10]: Use:

```
constructor(private exampleService: ExamplesService) {
    this.exampleService.loadExamples().subscribe(s => console.log(s));
}
```

for formatting `.ts` files.

Comment (Artfloriani) [reply src/app/modules/example-feature/components/presentation/presentation.component.ts:10]: I just updated this branch with master and now our linter throws an error on this kind of issues :) 

You can even view the errors on https://github.com/carbonaut-io/pwa-boilerplate/pull/14/files

Comment (lfa-vasconcellos) [reply src/app/modules/example-feature/components/presentation/presentation.component.ts]: Done

File: src/app/core/services/storage/storage.service.ts (hunk 5 above)
Comment (lfa-vasconcellos) [reply src/app/core/services/storage/storage.service.ts]: Fixed

**Pull request merged**